erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -1601,3 +1601,52 @@ def validate_sample_quantity(item_code, sample_quantity, qty, batch_no = None):
 			format(max_retain_qty, batch_no, item_code), alert=True)
 		sample_quantity = qty_diff
 	return sample_quantity
+
+def raw_material_update_on_bom():
+	"""
+	Calculates Average Manufactured Qty(avg_manufactured_qty) for BOM on the Basis of Stock Entries created and
+	it fetches the Stock Entries created within specified number of days in Stock Settings if not
+	specified then it calculates for past seven days against BOM.
+	"""
+	past_seven_days = get_date_str(add_days(today(), -7))
+	no_of_days = frappe.db.get_single_value("Stock Settings", "no_of_days_to_calculate_avg_manufactured_qty")
```

Comment:
```suggestion
	no_of_days = frappe.db.get_single_value("Stock Settings", "no_of_days_to_calculate_avg_manufactured_qty")
```

no_of_days should be used to calculate date. I cannot see where you are calculating the dates